forte/integrals/custom_integrals.h

```diff
@@ -96,6 +96,10 @@ class CustomIntegrals : public ForteIntegrals {
     void set_tei(size_t p, size_t q, size_t r, size_t s, double value, bool alpha1,
                  bool alpha2) override;
 
+    void set_tei_from_asints(std::shared_ptr<ActiveSpaceIntegrals> as_ints) override;
+
+    void set_ints_from_asints(std::shared_ptr<ActiveSpaceIntegrals> as_ints) override;
```

Comment:
I wonder if we could avoid these two functions. If we return the custom integrals in DSRG `compute_Heff_actv` and modify the Python side of DSRG, is it possible to eliminate these two?

Reply:
I prefer to keep them since if we do that, we need to turn DSRG into a polymorphous class similar to integrals.
We can have a discussion on this.

Reply:
I think the code should look like
```
# do first DSRG-MRPT2
dsrg = make_dsrg(...)
dsrg.compute_Heff_actv(mo_space_info_active)

# do second high-level DSRG
dsrg = make_dsrg(...)
dsrg.compute_Heff_actv()
```

Reply:
I think if we do this, "compute_Heff_actv" will need to do many additional works, which is not part of DSRG. And since active space relaxation also needs this function, I need to make it overloaded, it will again go back to a DSRG-embedding coupled code.

Reply:
I think I am missing something. Can't we just add another function, something like the following?
```
std::shared_ptr<ForteIntegrals> compute_Heff_actv_fints() {
// identical to the original compute_Heff_actv function except for the return type (std::shared_ptr<ActiveSpaceIntegrals>)
return make_custom_forte_integrals(options, mo_space_info, scalar,
                                   oei, oei, tei_aa, tei, tei_aa);
}
```
This seems the simplest solution I can think of.

The less ideal one would be add another constructor of `ForteIntegrals` using `ActiveSpaceIntegrals`.

The setting functions, in my opinion, might be the worst of these three options.

Reply:
I think I am missing something. Can't we just add another function, something like the following?
```
std::shared_ptr<ForteIntegrals> compute_Heff_actv_fints() {
    // identical to the original compute_Heff_actv function except for the return type
    // which is std::shared_ptr<ActiveSpaceIntegrals>

    // ... some copy of the original code

    // return
    return make_custom_forte_integrals(options, mo_space_info, scalar,
                                       oei, oei, tei_aa, tei, tei_aa);
}
```
This seems the simplest solution I can think of.

The less ideal one would be add another constructor of `ForteIntegrals` using `ActiveSpaceIntegrals`.

The setting functions, in my opinion, might be the worst of these three options.